lightning-invoice/src/lib.rs

```diff
@@ -1974,4 +1997,25 @@ mod test {
 
 		assert!(invoice.would_expire(Duration::from_secs(1234567 + DEFAULT_EXPIRY_TIME + 1)));
 	}
+
+	#[cfg(feature = "serde")]
+	#[test]
+	fn test_serde() {
+		let invoice = "lnbc100p1psj9jhxdqud3jxktt5w46x7unfv9kz6mn0v3jsnp4q0d3p2sfluzdx45tqcs\
+							h2pu5qc7lgq0xs578ngs6s0s68ua4h7cvspp5q6rmq35js88zp5dvwrv9m459tnk2zunwj5jalqtyxqulh0l\
+							5gflssp5nf55ny5gcrfl30xuhzj3nphgj27rstekmr9fw3ny5989s300gyus9qyysgqcqpcrzjqw2sxwe993\
+							h5pcm4dxzpvttgza8zhkqxpgffcrf5v25nwpr3cmfg7z54kuqq8rgqqqqqqqq2qqqqq9qq9qrzjqd0ylaqcl\
+							j9424x9m8h2vcukcgnm6s56xfgu3j78zyqzhgs4hlpzvznlugqq9vsqqqqqqqlgqqqqqeqq9qrzjqwldmj9d\
+							ha74df76zhx6l9we0vjdquygcdt3kssupehe64g6yyp5yz5rhuqqwccqqyqqqqlgqqqqjcqq9qrzjqf9e58a\
+							guqr0rcun0ajlvmzq3ek63cw2w282gv3z5uupmuwvgjtq2z55qsqqg6qqqyqqqrtnqqqzq3cqygrzjqvphms\
+							ywntrrhqjcraumvc4y6r8v4z5v593trte429v4hredj7ms5z52usqq9ngqqqqqqqlgqqqqqqgq9qrzjq2v0v\
+							p62g49p7569ev48cmulecsxe59lvaw3wlxm7r982zxa9zzj7z5l0cqqxusqqyqqqqlgqqqqqzsqygarl9fh3\
+							8s0gyuxjjgux34w75dnc6xp2l35j7es3jd4ugt3lu0xzre26yg5m7ke54n2d5sym4xcmxtl8238xxvw5h5h5\
+							j5r6drg6k6zcqj0fcwg";
+		let invoice = invoice.parse::<Invoice>().unwrap();
+		let serialized_invoice = serde_json::to_string(&invoice).unwrap();
+		let deserialized_invoice: Invoice = serde_json::from_str(serialized_invoice.as_str()).unwrap();
+		assert_eq!(invoice, deserialized_invoice);
+		assert_eq!(invoice.to_string(), deserialized_invoice.to_string());
```

Comment:
Could you also test here that the serialized invoice actually matches the initial string provided? Otherwise there might be the slight possibility that you make a mistake in the initial parsing and here only ever confirm that the logic is able to deserialize whatever you're serializing from that.

Reply:
But `serde_json::to_string(...)` serializes as JSON so the str is not "lnbc1..." but "\"lnbc1...\"" so I can't test that right ?

Reply:
Well, if it's indeed only about the wrapping `"`s, then you could just strip them before comparing the strings?

Reply:
Right, that works I'll force push again thank you